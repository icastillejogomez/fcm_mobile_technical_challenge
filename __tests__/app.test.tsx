import 'react-native'

import { renderRouter, screen } from 'expo-router/testing-library'

// Note: import explicitly to use the types shipped with jest.
import { it, describe } from '@jest/globals'

describe('When rendering the app', () => {
  it('should render the app', async () => {
    const app = renderRouter()

    expect(screen).toHavePathname('/places')
    app.getByText('Loading...')
    expect(screen).toMatchSnapshot()
  })
})
