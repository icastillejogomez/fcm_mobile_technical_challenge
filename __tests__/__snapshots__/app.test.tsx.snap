// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When rendering the app should render the app 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenNavigationContainer
        style={
          {
            "flex": 1,
            "overflow": "hidden",
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          onGestureCancel={[Function]}
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              [
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                  "flex": 1,
                },
                [
                  {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                  },
                  {
                    "backgroundColor": "#fff",
                  },
                ],
              ]
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  {
                    "flex": 1,
                  }
                }
              >
                <AIRMap
                  initialRegion={null}
                  onChange={[Function]}
                  onMapReady={[Function]}
                  region={null}
                  style={
                    {
                      "height": "100%",
                      "width": "100%",
                    }
                  }
                />
                <View
                  onLayout={[Function]}
                  pointerEvents="box-none"
                  style={
                    [
                      undefined,
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      {
                        "bottom": 0,
                        "overflow": "hidden",
                        "top": -60,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        undefined,
                        {
                          "flexDirection": "column-reverse",
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        {
                          "opacity": 1,
                          "transform": [
                            {
                              "translateY": 1334,
                            },
                          ],
                        },
                      ]
                    }
                  >
                    <View
                      accessibilityLabel="Bottom Sheet"
                      accessibilityRole="adjustable"
                      accessible={true}
                      pointerEvents="none"
                      style={
                        [
                          {
                            "backgroundColor": "white",
                            "borderRadius": 15,
                          },
                          {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          },
                        ]
                      }
                    />
                    <View
                      accessibilityLabel="Bottom Sheet"
                      accessibilityRole="adjustable"
                      accessible={true}
                      pointerEvents="box-none"
                      style={
                        [
                          {
                            "overflow": "hidden",
                          },
                          {
                            "paddingBottom": undefined,
                          },
                        ]
                      }
                    >
                      <View
                        collapsable={false}
                        handlerTag={1}
                        handlerType="PanGestureHandler"
                        onGestureHandlerEvent={[Function]}
                        onGestureHandlerStateChange={[Function]}
                        style={
                          [
                            {},
                            {
                              "height": NaN,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            [
                              {
                                "alignItems": "center",
                                "justifyContent": "center",
                                "marginBottom": 10,
                              },
                              {
                                "height": 60,
                              },
                            ]
                          }
                        >
                          <Text>
                            180 places to see
                          </Text>
                        </View>
                        <View
                          style={
                            {
                              "flex": 1,
                            }
                          }
                        >
                          <RCTScrollView
                            data={
                              [
                                {
                                  "data": [
                                    "Item 0",
                                    "Item 1",
                                    "Item 2",
                                    "Item 3",
                                    "Item 4",
                                    "Item 5",
                                    "Item 6",
                                    "Item 7",
                                    "Item 8",
                                    "Item 9",
                                  ],
                                  "title": "Section 0",
                                },
                                {
                                  "data": [
                                    "Item 0",
                                    "Item 1",
                                    "Item 2",
                                    "Item 3",
                                    "Item 4",
                                    "Item 5",
                                    "Item 6",
                                    "Item 7",
                                    "Item 8",
                                    "Item 9",
                                  ],
                                  "title": "Section 1",
                                },
                                {
                                  "data": [
                                    "Item 0",
                                    "Item 1",
                                    "Item 2",
                                    "Item 3",
                                    "Item 4",
                                    "Item 5",
                                    "Item 6",
                                    "Item 7",
                                    "Item 8",
                                    "Item 9",
                                  ],
                                  "title": "Section 2",
                                },
                                {
                                  "data": [
                                    "Item 0",
                                    "Item 1",
                                    "Item 2",
                                    "Item 3",
                                    "Item 4",
                                    "Item 5",
                                    "Item 6",
                                    "Item 7",
                                    "Item 8",
                                    "Item 9",
                                  ],
                                  "title": "Section 3",
                                },
                                {
                                  "data": [
                                    "Item 0",
                                    "Item 1",
                                    "Item 2",
                                    "Item 3",
                                    "Item 4",
                                    "Item 5",
                                    "Item 6",
                                    "Item 7",
                                    "Item 8",
                                    "Item 9",
                                  ],
                                  "title": "Section 4",
                                },
                                {
                                  "data": [
                                    "Item 0",
                                    "Item 1",
                                    "Item 2",
                                    "Item 3",
                                    "Item 4",
                                    "Item 5",
                                    "Item 6",
                                    "Item 7",
                                    "Item 8",
                                    "Item 9",
                                  ],
                                  "title": "Section 5",
                                },
                                {
                                  "data": [
                                    "Item 0",
                                    "Item 1",
                                    "Item 2",
                                    "Item 3",
                                    "Item 4",
                                    "Item 5",
                                    "Item 6",
                                    "Item 7",
                                    "Item 8",
                                    "Item 9",
                                  ],
                                  "title": "Section 6",
                                },
                                {
                                  "data": [
                                    "Item 0",
                                    "Item 1",
                                    "Item 2",
                                    "Item 3",
                                    "Item 4",
                                    "Item 5",
                                    "Item 6",
                                    "Item 7",
                                    "Item 8",
                                    "Item 9",
                                  ],
                                  "title": "Section 7",
                                },
                                {
                                  "data": [
                                    "Item 0",
                                    "Item 1",
                                    "Item 2",
                                    "Item 3",
                                    "Item 4",
                                    "Item 5",
                                    "Item 6",
                                    "Item 7",
                                    "Item 8",
                                    "Item 9",
                                  ],
                                  "title": "Section 8",
                                },
                                {
                                  "data": [
                                    "Item 0",
                                    "Item 1",
                                    "Item 2",
                                    "Item 3",
                                    "Item 4",
                                    "Item 5",
                                    "Item 6",
                                    "Item 7",
                                    "Item 8",
                                    "Item 9",
                                  ],
                                  "title": "Section 9",
                                },
                              ]
                            }
                            getItem={[Function]}
                            getItemCount={[Function]}
                            keyExtractor={[Function]}
                            onContentSizeChange={[Function]}
                            onLayout={[Function]}
                            onMomentumScrollBegin={[Function]}
                            onMomentumScrollEnd={[Function]}
                            onScroll={[Function]}
                            onScrollBeginDrag={[Function]}
                            onScrollEndDrag={[Function]}
                            renderItem={[Function]}
                            scrollEventThrottle={0.0001}
                            stickyHeaderIndices={
                              [
                                0,
                              ]
                            }
                            testID="mocked-flatlist"
                          >
                            <View>
                              <View
                                onFocusCapture={[Function]}
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    {
                                      "backgroundColor": "white",
                                      "padding": 6,
                                    }
                                  }
                                >
                                  <Text>
                                    Section 0
                                  </Text>
                                </View>
                              </View>
                              <View
                                onFocusCapture={[Function]}
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    {
                                      "backgroundColor": "#eee",
                                      "margin": 6,
                                      "padding": 6,
                                    }
                                  }
                                >
                                  <Text>
                                    Item 0
                                  </Text>
                                </View>
                              </View>
                              <View
                                onFocusCapture={[Function]}
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    {
                                      "backgroundColor": "#eee",
                                      "margin": 6,
                                      "padding": 6,
                                    }
                                  }
                                >
                                  <Text>
                                    Item 1
                                  </Text>
                                </View>
                              </View>
                              <View
                                onFocusCapture={[Function]}
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    {
                                      "backgroundColor": "#eee",
                                      "margin": 6,
                                      "padding": 6,
                                    }
                                  }
                                >
                                  <Text>
                                    Item 2
                                  </Text>
                                </View>
                              </View>
                              <View
                                onFocusCapture={[Function]}
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    {
                                      "backgroundColor": "#eee",
                                      "margin": 6,
                                      "padding": 6,
                                    }
                                  }
                                >
                                  <Text>
                                    Item 3
                                  </Text>
                                </View>
                              </View>
                              <View
                                onFocusCapture={[Function]}
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    {
                                      "backgroundColor": "#eee",
                                      "margin": 6,
                                      "padding": 6,
                                    }
                                  }
                                >
                                  <Text>
                                    Item 4
                                  </Text>
                                </View>
                              </View>
                              <View
                                onFocusCapture={[Function]}
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    {
                                      "backgroundColor": "#eee",
                                      "margin": 6,
                                      "padding": 6,
                                    }
                                  }
                                >
                                  <Text>
                                    Item 5
                                  </Text>
                                </View>
                              </View>
                              <View
                                onFocusCapture={[Function]}
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    {
                                      "backgroundColor": "#eee",
                                      "margin": 6,
                                      "padding": 6,
                                    }
                                  }
                                >
                                  <Text>
                                    Item 6
                                  </Text>
                                </View>
                              </View>
                              <View
                                onFocusCapture={[Function]}
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    {
                                      "backgroundColor": "#eee",
                                      "margin": 6,
                                      "padding": 6,
                                    }
                                  }
                                >
                                  <Text>
                                    Item 7
                                  </Text>
                                </View>
                              </View>
                              <View
                                onFocusCapture={[Function]}
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    {
                                      "backgroundColor": "#eee",
                                      "margin": 6,
                                      "padding": 6,
                                    }
                                  }
                                >
                                  <Text>
                                    Item 8
                                  </Text>
                                </View>
                              </View>
                              <View
                                style={
                                  {
                                    "height": 0,
                                  }
                                }
                              />
                            </View>
                          </RCTScrollView>
                        </View>
                      </View>
                    </View>
                    <View
                      collapsable={false}
                      handlerTag={2}
                      handlerType="PanGestureHandler"
                      onGestureHandlerEvent={[Function]}
                      onGestureHandlerStateChange={[Function]}
                      onLayout={[Function]}
                    >
                      <View
                        accessibilityHint="Drag up or down to extend or minimize the bottom sheet"
                        accessibilityLabel="Bottom sheet handle"
                        accessibilityRole="adjustable"
                        accessible={true}
                        style={
                          [
                            {
                              "padding": 10,
                            },
                            [
                              undefined,
                            ],
                          ]
                        }
                      >
                        <View
                          style={
                            [
                              {
                                "alignSelf": "center",
                                "backgroundColor": "rgba(0, 0, 0, 0.75)",
                                "borderRadius": 4,
                                "height": 4,
                                "width": 56.25,
                              },
                              [
                                undefined,
                              ],
                            ]
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  [
                    {
                      "borderBottomWidth": 1,
                      "gap": 8,
                    },
                    {
                      "backgroundColor": "#fff",
                      "borderBottomColor": "#C8C8C8",
                      "paddingHorizontal": 16,
                      "paddingTop": 0,
                    },
                  ]
                }
              >
                <View
                  style={
                    {
                      "borderColor": "red",
                      "borderWidth": 1,
                      "height": 40,
                    }
                  }
                >
                  <View
                    style={{}}
                  />
                </View>
                <View
                  style={
                    {
                      "borderColor": "blue",
                      "borderWidth": 1,
                      "height": 60,
                    }
                  }
                >
                  <View
                    style={{}}
                  />
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenNavigationContainer>
      <RCTSafeAreaView
        style={
          [
            {
              "borderTopWidth": 1,
              "flexDirection": "row",
            },
            {
              "backgroundColor": "#fff",
              "borderColor": "#C8C8C8",
              "height": 85,
            },
          ]
        }
      >
        <View
          style={
            {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": true,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "gap": 8,
                "justifyContent": "center",
                "opacity": 1,
                "paddingTop": 12,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 20,
                  "width": 20,
                }
              }
              tintColor="#00de00"
            />
            <Text
              style={
                {
                  "color": "#11181C",
                  "fontSize": 10,
                  "textAlign": "center",
                }
              }
            >
              Explore
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "gap": 8,
                "justifyContent": "center",
                "opacity": 1,
                "paddingTop": 12,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 20,
                  "width": 20,
                }
              }
              tintColor="#222"
            />
            <Text
              style={
                {
                  "color": "#222",
                  "fontSize": 10,
                  "textAlign": "center",
                }
              }
            >
              Bookmarks
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "gap": 8,
                "justifyContent": "center",
                "opacity": 1,
                "paddingTop": 12,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 20,
                  "width": 20,
                }
              }
              tintColor="#222"
            />
            <Text
              style={
                {
                  "color": "#222",
                  "fontSize": 10,
                  "textAlign": "center",
                }
              }
            >
              Bookings
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "gap": 8,
                "justifyContent": "center",
                "opacity": 1,
                "paddingTop": 12,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 20,
                  "width": 20,
                }
              }
              tintColor="#222"
            />
            <Text
              style={
                {
                  "color": "#222",
                  "fontSize": 10,
                  "textAlign": "center",
                }
              }
            >
              Messages
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "gap": 8,
                "justifyContent": "center",
                "opacity": 1,
                "paddingTop": 12,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 20,
                  "width": 20,
                }
              }
              tintColor="#222"
            />
            <Text
              style={
                {
                  "color": "#222",
                  "fontSize": 10,
                  "textAlign": "center",
                }
              }
            >
              Profile
            </Text>
          </View>
        </View>
      </RCTSafeAreaView>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
